src/main/java/wedlog/address/model/person/GuestRsvpPredicate.java

```diff
@@ -0,0 +1,47 @@
+package wedlog.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import wedlog.address.commons.util.StringUtil;
+import wedlog.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Guest}'s {@code Rsvp} matches any of the keywords given.
+ */
+public class GuestRsvpPredicate implements Predicate<Guest> {
+    private final List<String> keywords;
+
+    /**
+     * Constructor for GuestRsvpPredicate.
+     */
+    public GuestRsvpPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Guest guest) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(guest.getRsvpStatus().value, keyword));
+    }
```

Comment:
I was thinking that for the rest of the parameters, an empty input would filter for guests that do not have that parameter filled. with that in mind, i am of the opinion that it would be a bit confusing for the user to think back to another command with a different format. We could possibly bring this up for discussion?

Reply:
In some sense, it is not really different, because an rsvp status of type `unknown` is equivalent to other parameters which are not filled. I won't touch the implementation, I do agree we can bring this up for discussion.

Reply:
I think its ok to leave it as "unknown" because the UI represents a missing/unknown RSVP status as such.